### PR TITLE
Migrate /emoteonly and /emoteonlyoff commands to the Helix API

### LINKED ISSUE #3966: Migrate /emoteonly command to Helix API
As part of Twitch's announced deprecation of IRC-based commands ([see here for more info](https://discuss.dev.twitch.tv/t/deprecation-of-chat-commands-through-irc/40486), the `/emoteonly` command needs to be migrated to use the relevant Helix API endpoint.

Helix API reference: https://dev.twitch.tv/docs/api/reference#update-chat-settings

Split from #3956

### PR DESCRIPTION
**Pull request by Nerixyz**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

This is the first part of updating commands that use the _Update Chat Settings_ endpoint.

Fixes #3966, #3967.

### CODE REVIEW

File: src/controllers/commands/CommandController.cpp [hunk 1]
```diff
@@ -1725,6 +1725,119 @@ void CommandController::initialize(Settings &, Paths &paths)
...
+                channel->addMessage(makeSystemMessage(
+                    "You must be logged in to update chat settings!"));
+                return "";
+            }
+            auto *twitchChannel = dynamic_cast<TwitchChannel *>(channel.get());
+            if (twitchChannel == nullptr)
+            {
+                channel->addMessage(makeSystemMessage(
+                    "The /emoteonlyoff command only works in Twitch channels"));
+                return "";
```
Comment (Nerixyz) [inline src/controllers/commands/CommandController.cpp:1820]: This is almost the same as in other twitch commands. In my opinion you should be able to do something like this:

```cpp
this->registerTwitchCommand("/xd", 
                            [](const QStringList&,
                               std::shared_ptr<TwitchChannel>,
                               std::shared_ptr<TwitchAccount>) { 
// ...
});
```

**Review by Felanbird — DISMISSED** [review]
nitpicks

File: src/providers/twitch/api/Helix.hpp [hunk 2]
```diff
@@ -391,6 +421,16 @@ enum class HelixAddChannelVIPError {
...
 };
 
+enum class HelixUpdateChatSettingsError {
+    Unknown,
+    UserMissingScope,
+    UserNotAuthorized,
+    Ratelimited,
+    Forbidden,
+    // The error message is forwarded directly from the Twitch API
+    Forwarded,
```
[the remark this answers is not in the document]
Comment (Nerixyz) [reply src/providers/twitch/api/Helix.hpp:431]: All other errors have the comment above as well, so I kept it like this.

File: src/providers/twitch/api/Helix.cpp [hunk 3]
```diff
@@ -1257,6 +1257,169 @@ void Helix::addChannelVIP(
...
+    boost::optional<int> nonModeratorChatDelayDuration,
+    ResultCallback<HelixChatSettings> successCallback,
+    FailureCallback<HelixUpdateChatSettingsError, QString> failureCallback)
+{
+    QJsonObject json;
+    json["non_moderator_chat_delay"] =
+        nonModeratorChatDelayDuration.has_value();
+    if (nonModeratorChatDelayDuration)
+    {
+        json["slow_mode_wait_time"] = *nonModeratorChatDelayDuration;
```
Comment (pajlada) [inline src/providers/twitch/api/Helix.cpp:1299]: copy paste error?

**Review by pajlada — APPROVED** [review]

**Pull request merged**